src/third/redux-persist/persistStore.js

```diff
@@ -120,25 +129,25 @@ export default function persistStore (store, config = {}, onComplete) {
     store.dispatch({
       // Include a random string, to be sure no other action type
       // matches this one. Like Redux does for the initial action.
-      type: `PERSIST_DUMMY/${Math.floor(Math.random() * 2 ** 54).toString(36)}`
+      type: `PERSIST_DUMMY/${Math.floor(Math.random() * 2 ** 54).toString(36)}`,
     });
 
-    onComplete && onComplete(err, restoredState)
+    onComplete(err, restoredState);
```

Comment:
OTOH for `persistStore` (by contrast with `getStoredState` as discussed above), a callback makes sense because the callback is just an incidental hook, not the main point of the function.

Here's the callback we end up passing as this `onComplete`, from `StoreProvider.js`:
```js
    timing.start('Store hydration');
    restore(() => {
      timing.end('Store hydration');
    });
```

Given that, I think the appropriate API is for this one to stay optional. That timing is harmless and cheap, but something we rarely use; it's easy to imagine we might cut it out sometime in the future, and then it'd look silly to pass a no-op callback.

I think that means the net for this commit:

    redux-persist [nfc]: Assume some on-complete callbacks will be passed.

is:
* We'll drop it for now, i.e. from this PR.
* After adding types, we'll do a version of it that's just for `getStoredState`, and makes it never take a callback and always return a `Promise`.